mesonbuild/compilers/mixins/clike.py

```diff
@@ -727,24 +727,29 @@ def has_function(self, funcname, prefix, env, *, extra_args=None, dependencies=N
         # need to look for them differently. On nice compilers like clang, we
         # can just directly use the __has_builtin() macro.
         fargs['no_includes'] = '#include' not in prefix
-        fargs['__builtin_'] = '' if funcname.startswith('__builtin_') else '__builtin_'
+        is_builtin = funcname.startswith('__builtin_')
+        fargs['is_builtin'] = is_builtin
+        fargs['__builtin_'] = '' if is_builtin else '__builtin_'
         t = '''{prefix}
         int main(void) {{
+
+        /* With some toolchains (MSYS2/mingw for example) the compiler
+         * provides various builtins which are not really implemented and
```

Comment:
Just for the record (commenting on the comment, even if it was here before) - this isn't really toolchain specific - but mingw is maybe one of the more common platforms where more functions are missing (hence the need for build system checks). The compilers provide their builtins regardless of what happens to be available in the platform headers (the compiler doesn't know what, and platform headers evolve independently from compilers).

Reply:
Thanks, good to know. The previous comment I replaced here said the toolchain is broken, and I wasn't sure what to blame.. :)